main.cpp

```diff
@@ -46,20 +46,75 @@ Project 4: Part 4 / 9
  You will need to use Forward Declaration and out-of-class definitions to complete this.
  */
 
+#include "IntType.h"
+#include "FloatType.h"
+#include "DoubleType.h"
 
 
 struct Point
 {
+    Point(DoubleType& x_, DoubleType& y_);
+    Point(IntType& x_, IntType& y_);
+    Point(FloatType& x_, FloatType& y_);
+    Point(float x_, float y_) : x { x_} , y { y_} {}
+
     Point& multiply(float m)
     {
         x *= m;
         y *= m;
         return *this;
     }
+
+    Point& multiply(DoubleType& m)
+    {
+        x *= static_cast<float>(m);
+        y *= static_cast<float>(m);
+        return *this;
+    }
+
+    Point& multiply(IntType& m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+
+    Point& multiply(FloatType& m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
```

Comment:
All of these should call the version that takes a primitive.  And they should all take their param by const ref, not ref.
- [ ] revise.